### ISSUE TITLE
Geant4 exception when simulating taus

### ISSUE THREAD

**Issue by gaede** [AIDASoft/DD4hep#383]
- OS version: SLC6, MacOS, ...
 - Compiler version: e.g GCC 4.9, 7.3 llvm 9.1.0 (clang-902.0.39.1)
 - DD4hep version: current HEAD, v01-07
 - Reproduced by, for example: 
       - apply this patch:  https://github.com/AIDASoft/DD4hep/pull/381/files
          (use taus rather then electrons in sim test)
      - build DD4hep examples (at least CLICSiD)
      - run `ctest -V -R CLICSiDXML`
     - this will eventually produce an error message similar to:
```
13: ParticleHandler  INFO  +++ Event 83 Begin event action. Access event related information.
13: G4EnergyRangeManager:GetHadronicInteraction: counter=2, Ek=4.0100440938635e+77, Material = CarbonFiber, Element = HYDROGEN
13: *0* low=3000, high=100000000
13: *1* low=0, high=12000
13: In /data/ilcsoft/geant4/10.03.p02/source/processes/hadronic/management/src/G4EnergyRangeManager.cc, line 129: 
13: ===> GetHadronicInteraction: No Model found
13: 
13: -------- EEEE ------- G4Exception-START -------- EEEE -------
13: *** G4Exception : had005
13:       issued by : G4HadronicProcess::PostStepDoIt
13: In /data/ilcsoft/geant4/10.03.p02/source/processes/hadronic/management/src/G4EnergyRangeManager.cc, line 129: 
13: ===> GetHadronicInteraction: No Model found
13: Target element HYDROGEN  Z= 1  A= 1
13: Unrecoverable error in the method ChooseHadronicInteraction of pi-Inelastic
13: TrackID= 3  ParentID= 1  pi-
13: Ekin(GeV)= 4.01004e+74;  direction= (0.866884,0.479945,0.134777)
13: Position(mm)= (147.584,81.7223,22.9702);  material CarbonFiber
13: PhysicalVolume  <VertexBarrelSupports_layer0_slice0_0>
13:  No HadronicInteraction found out
13: 
13: *** Fatal Exception *** core dump ***
13: -------- EEEE -------- G4Exception-END --------- EEEE -------
```

 - as reported in the last meeting, this problem was originally observed in the currently running ILD mass production (using stdhep files, used before in a large scale production)
- since then I could reproduce this by using a simple particle gun with taus in various detector models  (ILD_l5_v02, CLIC_o3_v14) - also using different versions of geant4 (10.2.2, 10.3.2, 10.3.3)
- in order to exclude any issue in the LC detector models or ddsim, I made the above PR that suns a standard DD4hep test
- not sure if this is a DD4hep/DDG4 problem or a Geant4 problem really
      - just start the issue here to keep track of it ....

**Comment by gaede** [comment]
This seems to be indeed a Geant4 bug - I was able to reproduce it in a standalone Geant4 example (basic/B2a) with geant4 10.3.2...

**Comment by andresailer** [comment]
@cgrefe has ATLAS or have you seen this exception in the simulation of taus?

**Comment by cgrefe** [comment]
@andresailer no I haven't. Our MC simulation is run centrally, so I don't know if this has been observed or even fixed.

**Comment by gaede** [comment]
I have submitted a bug report with geant =4 - let's see if they can help:
https://bugzilla-geant4.kek.jp/show_bug.cgi?id=2061

**Comment by petricm** [comment]
Fixed in Geant4 10.04.p02
